ksqldb-engine/src/main/java/io/confluent/ksql/embedded/KsqlContext.java

```diff
@@ -189,7 +189,10 @@ public void close() {
   }
 
   public void terminateQuery(final QueryId queryId) {
```

Comment:
It looks like this actually isn't being called from anywhere outside of test code...at least that's what my IDE is saying. Seems like a problem? Not sure where/how we actually terminate queries, or why we have this method if it's only for tests -- maybe should be marked with `@VisibleForTesting` since it's very much not clear that it isn't used  for "real" queries 😬 

Was poking around because it looks like we aren't handling the CREATE OR REPLACE case properly for https://github.com/confluentinc/ksql/pull/8538, ie we need to remove the old source topics from the RuntimeAssignor when a query is replaced -- we should just need to make sure that we invoke `#dropQuery` and then re-register that query in the `RuntimeAssignor`. Could you look into this?

Reply:
If you want to test it out, I added a CREATE OR REPLACE test to EndToEndIntegrationTest in my PR, plus you'll need [this fix] (https://github.com/confluentinc/ksql/pull/8538/files#diff-28490c7f037eb219dd2e44fabbf194d2d8805f895becd6daf48f79bf3ce4f09aR123-R126)for a bug that was causing RuntimeAssignor#stopQuery to throw an exception. Although given that I found the bug in #stopQuery at all, when this is only called from testing code, makes me wonder how useful those integration tests really are for replicating the real thing...

Reply:
For now this isn't being used. This PR is what is using this method. Until it is merged then that won't be used. see the file below.

Reply:
Sorry I meant `terminateQuery` not `stopQuery`